core/src/main/java/org/testcontainers/containers/startupcheck/StartupCheckStrategy.java

```diff
@@ -0,0 +1,51 @@
+package org.testcontainers.containers.startupcheck;
+
+import com.github.dockerjava.api.DockerClient;
+import com.github.dockerjava.api.command.InspectContainerResponse;
+import org.rnorth.ducttape.ratelimits.RateLimiter;
+import org.rnorth.ducttape.ratelimits.RateLimiterBuilder;
+import org.rnorth.ducttape.unreliables.Unreliables;
+
+import java.util.concurrent.TimeUnit;
+
+import static org.testcontainers.containers.GenericContainer.CONTAINER_RUNNING_TIMEOUT_SEC;
+
+/**
+ * Approach to determine whether a container has 'started up' correctly.
+ */
+public abstract class StartupCheckStrategy {
```

Comment:
@ahaines: I've refactored startup checks somewhat in this branch - I wonder if this might help with #154